src/main/perl/FileEditor.pm

```diff
@@ -58,13 +58,38 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
+{
+    my ($self) = @_;
+    my $is_newer = 0;   # Assume false
+    if ( $self->_is_valid_source(*$self->{options}->{source}) ) {
+        if ( !$self->_is_valid_source(*$self->{filename}) || 
+             ((stat(*$self->{options}->{source}))[9] > (stat(*$self->{filename}))[9]) ) {
+          $is_newer = 1
+        }
+    }
+    if ( $is_newer ) {
+        *$self->{LOG}->debug(2, "Source file (", *$self->{options}->{source}, " is newer than ", *$self->{filename}, ": use it");
+    } else {
+        *$self->{LOG}->debug(2, "Source file (", *$self->{options}->{source}, " older than ", *$self->{filename}, ": ignore it");
+    };
+    return $is_newer;
+}
 
 sub new
 {
     my $class = shift;
     my $self = $class->SUPER::new (@_);
-    if ($self->_is_valid_source(*$self->{filename})) {
-        my $txt = LC::File::file_contents (*$self->{filename});
+    my $txt;
+    my ($path, %opts) = @_;
+
+    *$self->{options}->{source} = $opts{source} if exists ($opts{source});
+    if ( exists(*$self->{options}->{source}) && $self->_is_reference_newer() ) {
+        $txt = LC::File::file_contents (*$self->{options}->{source});
```

Comment:
also avoid code duplication. best to only modify the filename to read from, and use the `LC::File::file_contents`
even better, let `_is_reference_newer` return the correct filename to read from